test/integration/integration_test.go

```diff
@@ -725,11 +736,57 @@ func succeedsOrFails(execute commandRunner, command string, expectedResult strin
 	return nil
 }
 
-func verifyHTTPResponse(partOfResponse string, url string, urlSuffix string, assertion string, expected string) error {
-	switch url {
-	case "OpenShift":
-		url = minishift.getOpenShiftUrl() + urlSuffix
+func verifyRequestToURL(partOfResponse string, url string, assertion string, expected string) error {
+	return verifyHTTPResponse(partOfResponse, url, assertion, expected)
+}
+
+func verifyRequestToURLWithRetry(retryCount int, retryMaxOverallTime int, partOfResponse string, url string, assertion string, expected string) error {
+	return verifyHTTPResponseWithRetry(partOfResponse, url, assertion, expected, retryCount, retryMaxOverallTime)
+}
+
+func verifyRequestToService(partOfResponse string, urlSuffix string, serviceName string, nameSpace string, assertion string, expected string) error {
+	url := minishift.getRoute(serviceName, nameSpace) + urlSuffix
+	return verifyHTTPResponse(partOfResponse, url, assertion, expected)
+}
+
+func verifyRequestToServiceWithRetry(retryCount int, retryMaxOverallTime int, partOfResponse string, urlSuffix string, serviceName string, nameSpace string, assertion string, expected string) error {
+	url := minishift.getRoute(serviceName, nameSpace) + urlSuffix
+	return verifyHTTPResponseWithRetry(partOfResponse, url, assertion, expected, retryCount, retryMaxOverallTime)
+}
+
+func verifyRequestToOpenShift(partOfResponse string, urlSuffix string, assertion string, expected string) error {
+	url := minishift.getOpenShiftUrl() + urlSuffix
+	return verifyHTTPResponse(partOfResponse, url, assertion, expected)
+}
+
+func verifyRequestToOpenShiftWithRetry(retryCount int, retryMaxOverallTime int, partOfResponse string, urlSuffix string, assertion string, expected string) error {
+	url := minishift.getOpenShiftUrl() + urlSuffix
+	return verifyHTTPResponseWithRetry(partOfResponse, url, assertion, expected, retryCount, retryMaxOverallTime)
+}
+
+func verifyHTTPResponseWithRetry(partOfResponse string, url string, assertion string, expected string, retryCount int, retryMaxOverallTime int) error {
+	var err error
+	divisor := (retryCount * (retryCount + 1)) / 2
+
+	for i := 0; i <= retryCount; i++ {
+		err = verifyHTTPResponse(partOfResponse, url, assertion, expected)
+		if err == nil {
+			break
+		}
+		if i == retryCount {
+			fmt.Printf("HTTP check (%v) has failed. Error: %v\n", i+1, err)
+			break
+		}
+
+		delay := (i + 1) * retryMaxOverallTime / divisor
+		fmt.Printf("HTTP check (%v) has failed, trying again in %v ms. Error: %v\n", i+1, delay, err)
+		time.Sleep(time.Millisecond * time.Duration(delay))
```

Comment:
@agajdosi I think we would be fine if the the unit is second. Millisecond is too granular.

Reply:
@LalatenduMohanty I agree seconds are friendlier but here we are really handling processes which happen it time frame of milliseconds. But if needed I can add a converter so it can be defined in both seconds and milliseconds.